src/NuGet.Core/NuGet.Packaging/PackageArchiveReader.cs

```diff
@@ -228,6 +228,24 @@ public IEnumerable<ZipFilePair> EnumeratePackageEntries(IEnumerable<string> pack
             }
         }
 
+        /// <summary>
+        /// Validate all files in package are not traversed outside of the expected extraction path.
+        /// Eg: file entry like ../../foo.dll can get outside of the expected extraction path.
+        /// </summary>
+        public async Task<bool> ValidatePackageEntriesAsync(CancellationToken token)
+        {
+            var files = await GetFilesAsync(token);
+            // This dumy destionation is used to check if the file in package get outside of the extractionPath
+            var dumyDestination = Path.GetFullPath(Path.Combine(".nuget", "packages"));
+
+            return files.All(p =>
+            {
+                var normalizedPath = Uri.UnescapeDataString(p.Replace('/', Path.DirectorySeparatorChar));
+                var fullPath = Path.GetFullPath(Path.Combine(dumyDestination, normalizedPath));
```

Comment:
The [ZIP specification](https://pkware.cachefly.net/webdocs/casestudies/APPNOTE.TXT) allows entry file names to be up to 65535 characters long.  Given the entry file name plus [what you're tacking on to the front](https://github.com/NuGet/NuGet.Client/pull/2296/files#diff-f78aa5e2a43496c45e31b8660023d78dR239), this statement can throw a `System.IO.PathTooLongException`.

In general, I think you need to handle all exceptions that [`Path.GetFullPath(...)`](https://msdn.microsoft.com/en-us/library/system.io.path.getfullpath(v=vs.110).aspx) can throw.